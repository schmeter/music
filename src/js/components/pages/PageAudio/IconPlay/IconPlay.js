import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../../../Icon';

class IconPlay extends React.Component {
    render() {
        const { isPlaying } = this.props;

        return <Icon id={isPlaying ? 'pause' : 'play'} />;
    }
}

IconPlay.propTypes = {
    isPlaying: PropTypes.bool.isRequired,
};

export default IconPlay;
