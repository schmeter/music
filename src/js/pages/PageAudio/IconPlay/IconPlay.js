import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../../../components/Icon';

const IconPlay = ({
  isPlaying,
}) => (
  <Icon id={isPlaying ? 'pause' : 'play'} />
);

IconPlay.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
};

export default IconPlay;
