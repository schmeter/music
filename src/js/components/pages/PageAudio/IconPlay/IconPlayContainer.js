import { connect } from 'react-redux';

import IconPlay from './IconPlay';

const mapStateToProps = (state) => ({
    isPlaying: state.audio.isPlaying
});

export default connect(mapStateToProps, null)(IconPlay);
