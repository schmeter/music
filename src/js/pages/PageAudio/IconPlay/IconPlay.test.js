import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import IconPlay from './IconPlay';

jest.mock('../../../components/Icon', () => () => <div />);

describe('IconPlay', () => {
    it('renders correctly with isPlaying', () => {
        const props = {
            isPlaying: true,
        };
        const component = shallow(<IconPlay {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('renders correctly without isPlaying', () => {
        const props = {
            isPlaying: false,
        };
        const component = shallow(<IconPlay {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
