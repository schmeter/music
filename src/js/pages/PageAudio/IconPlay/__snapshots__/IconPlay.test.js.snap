// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconPlay renders correctly with isPlaying 1`] = `
<Component
  id="pause"
/>
`;

exports[`IconPlay renders correctly without isPlaying 1`] = `
<Component
  id="play"
/>
`;
